xgboost_ray/util.py

```diff
@@ -37,8 +39,12 @@ def shutdown(self):
             ray.kill(self.actor)
         self.actor = None
 
-
+# TODO: Propogate this into RayQueue.
 class Queue(RayQueue):
+    def __init__(self, maxsize: int = 0, actor_options: Dict = {}) -> None:
+        self.maxsize = maxsize
+        self.actor = _QueueActor.options(**actor_options).remote(self.maxsize)
+
```

Comment:
I think we could actually consolidate both functionalities into one, e.g. create something like a `RayCommunication` actor class which has an `asyncio.Event` and `asyncio.Queue` and provides the interface for both.

Reply:
(Maybe out of scope for this PR though)

Reply:
Yes that's a great idea- we should do that. I raised an issue to keep track of this.